app/domain/authentication/authn_jwt/jwt_fetch_jwks_uri_signing_key.rb

```diff
@@ -0,0 +1,51 @@
+require 'uri'
+require 'net/http'
+
+module Authentication
+  module AuthnJwt
+
+    JwtFetchJwksUriSigningKey = CommandClass.new(
+      dependencies: {
+        logger: Rails.logger,
+        discover_identity_provider: OAuth::DiscoverIdentityProvider.new,
+        fetch_secrets: Conjur::FetchRequiredSecrets.new
+      },
+      inputs: %i[authenticator_input]
+    ) do
+      def call
+        jwks_uri
+        fetch_jwks_keys
+      end
+
+      private
+
+      def jwks_uri
+        @logger.debug(LogMessages::Authentication::AuthnJwt::JwksUriResourceNameConfiguration.new(resource_id))
+        @jwks_uri_secret ||= @fetch_secrets.(resource_ids: [resource_id])
+        @jwks_uri ||=@jwks_uri_secret[resource_id]
+      end
+
+      def resource_id
+        "#{@authenticator_input.account}:variable:conjur/#{@authenticator_input.authenticator_name}/#{@authenticator_input.service_id}/#{JWKS_URI_RESOURCE_NAME}"
+      end
+
+      def fetch_jwks_keys
+        uri = URI(@jwks_uri)
```

Comment:
does `URI` need to be a dependency? would we want to mock it?

Reply:
also - please use `jwks_uri` and not `@jwks_uri`. we have a method that memoizes this object so calling it with the `@` can cause unexpected behaviour.